plugins/faustwp/includes/replacement/functions.php

```diff
@@ -131,3 +131,80 @@ function is_wp_link_ajax_request(): bool {
 		&& ! empty( $_POST['action'] )
 		&& 'wp-link-ajax' === $_POST['action'] );
 }
+
+/**
+ * Get all site URLs for each possible HTTP protocol
+ *
+ * @return array<string> An array of site urls.
+ */
+function faustwp_get_wp_site_urls() {
+
+	$site_url = site_url();
+	$host_url = wp_parse_url( $site_url, PHP_URL_HOST );
+
+	$is_https = strpos( $site_url, 0, 6 ) === 'https:';
+
+	return apply_filters(
+		'faustwp_get_wp_site_urls',
+		array(
+			$is_https ? "https://$host_url" : "http://$host_url",
+			$is_https ? "http://$host_url" : "https://$host_url",
+			"//$host_url",
+		)
+	);
+}
+
+/**
+ * Get all media urls based off the available site urls
+ *
+ * @param array<string> $wp_site_urls The array of potential site urls.
+ *
+ * @return array<string> The array of media Urls
+ */
+function faustwp_get_wp_media_urls( array $wp_site_urls ) {
+	$upload_url = faustwp_get_relative_upload_url( $wp_site_urls );
```

Comment:
I think we can just pass the `$upload_url` here as a parameter, then we won't need to call it once again inside the function.